website/docs/plugin/framework/diagnostics.mdx

```diff
@@ -245,12 +245,55 @@ func (s exampleType) Validate(ctx context.Context, in tftypes.Value, path path.P
     // ... further logic ...
 ```
 
-#### Custom Diagnostics Types
+### Consistent Diagnostic Creation
 
-Advanced provider developers may need to further differentiate or store
-additional data in error and warning types. The
-[`diag.Diagnostic` type](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/diag#Diagnostic)
-is an interface type that can be implemented with these methods:
+Create a helper function in your provider code to generate consistent diagnostics for types of errors/warnings. It is also possible to use [custom diagnostics types](#custom-diagnostics-types) to accomplish this same goal.
```

Comment:
> _Create a helper function in your provider code to generate consistent diagnostics for types of errors/warnings._

This first line reads like a recommendation to create your own helper function, but the following content describes the existing helpers we provide. Could restructure like: "you can utilize X helper functions we provide (list functions), or you can write your own custom helper function like [this example](https://github.com/hashicorp/terraform-plugin-framework/pull/687/files#diff-288f191e52a4d5293f97d279c6253ea1770f5b0ae793e6ad1c9f6b5d49d9fedeR261-R272)"

Reply:
Happy to adjust this -- just not sure what the best verbiage might be. In the code example below, `APIErrorDiagnostic()` is what I was intending to convey as the "create a helper function" part here. Is this suggested change maybe better here?

```suggestion
Create a helper function in your provider code using the diagnostic creation functions available in the [`diag` package](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/diag) to generate consistent diagnostics for types of errors/warnings. It is also possible to use [custom diagnostics types](#custom-diagnostics-types) to accomplish this same goal.
```

Reply:
Ah, I understand the intention now, that change clears it up for me! 👍🏻